### PR TITLE
feat: Hide dashboard app name DHIS2-9604

### PR DESCRIPTION
**Pull request by jimgrace**

### CODE REVIEW

**Review by larshelge — CHANGES_REQUESTED** [review]
Is this a property that applies to _all_ custom apps? 

What does "hide dashboard title" actually imply here?

Should we include specific properties (dashboard title) in the general app manifest?

**Comment by jimgrace** [comment]
@larshelge 

1. This would be an optional property for any dashboard app, custom or not. It would not need to be present, and if it is absent everything would work the way it does now.

2. It means that when the Dashboard app displays the 'dashboard app', it does not print the title at the top of the app. What PEPFAR wants to do is create a dashboard app that operates like the text box widget but can display formatted text in the box. They want only the text to appear in the dashboard item (like with the text widget), not also the app title.

3. Yes, PEPFAR would like the core dashboard app to optionally suppress printing the title of an app of type dashboard it is displaying, if it is directed to by the manifest.

4. This change isn't about putting a title or not in the app manifest, but putting a boolean option to hide the title or not -- when displaying a dashboard app on the dashboard. For the discussion of where to put this boolean in the manifest I was advised to consult with the frontend team, which you can see at https://dhis2.slack.com/archives/C0BP0RABF/p1586960401171800 (DHIS2 Slack, not public).

See also https://jira.dhis2.org/browse/DHIS2-9604 for this backend feature and https://jira.dhis2.org/browse/DHIS2-7660 for the overall feature.

**Comment by larshelge** [comment]
Okay thanks for the response. I understand now that this is about custom Dashboard _widgets_, I thought it applied to regular apps. I think this will be confusing to developers in the future as well, and we should try to make it clearer.

If we are starting to add properties to the manifest which applies to only certain app types I think we should qualify the properties with a "namespace" / object wrapper. The format could be something like this:

```json
{
  "settings": {
    "app": {
      },
    "resource": {
      },
    "dashboardWidget: {
      "hideTitle": true
      },
    "trackerDashboardWidget: {
      }
  }
}
```

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/appmanager/AppSettingsDashboardWidget.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ */
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+import org.hisp.dhis.common.DxfNamespaces;
+
+/**
+ * @author Jim Grace
+ */
+public class AppSettingsDashboardWidget
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/appmanager/AppSettingsDashboardWidget.java:38]: Could `DashboardWidgetAppSettings` be a more natural name?

File: dhis-2/dhis-api/src/test/resources/manifest.webapp [hunk 2]
```diff
@@ -38,5 +38,10 @@
       "APP_UPDATE",
       "APP_DELETE",
       "APP_ALLOW_EXECUTE"
-    ]
-}
\ No newline at end of file
+    ],
+    "settings": {
```
Comment (larshelge) [inline dhis-2/dhis-api/src/test/resources/manifest.webapp:42]: This format looks good to me :+1:

**Review by varl — APPROVED** [review]
Hi,

As discussed on Slack, we have plans to address this in a different way in the App Platform, but we have to do it in order, and that will take some time. Adding the issues for reference:

- https://github.com/dhis2/app-platform/issues/379 outlines the overall approach and steps.
- https://github.com/dhis2/app-platform/issues/375 is the issue tracking the "chromeless applications" (no headerbar, no title).

Approving this feature addition until then. :+1:

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/appmanager/AppSettingsDashboardWidget.java (hunk 1 above)
Comment (jimgrace) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/appmanager/AppSettingsDashboardWidget.java]: 100% right @larshelge, thanks. Renamed now. I was trying to follow the JSON path settings.dashboardWidget, and also put the class right after AppSettings in alphabetical order. But you are right that's the more natural name. Thanks. :)

**Review by larshelge — APPROVED** [review]

**Pull request merged**